modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/upload/TempAttachmentsUploadFileEntryHandler.java

```diff
@@ -61,11 +62,13 @@ public FileEntry upload(UploadPortletRequest uploadPortletRequest)
 
 		String fileName = uploadPortletRequest.getFileName(_PARAMETER_NAME);
 
-		_validateFile(fileName, uploadPortletRequest.getSize(_PARAMETER_NAME));
-
 		String contentType = uploadPortletRequest.getContentType(
 			_PARAMETER_NAME);
 
+		_validateFile(
+			fileName, uploadPortletRequest.getSize(_PARAMETER_NAME),
+			contentType);
+
```

Comment:
contentType needs to be first. After fix is made. you can close this PR and send directly to liferay-commerce. Assign ticket to Andrea Sbarra.